ds3_java_cli/src/main/java/com/spectralogic/ds3cli/command/PutBulk.java

```diff
@@ -371,6 +386,21 @@ public PipeFileObjectPutter(final ImmutableMap<String, String> mapNormalizedObje
         public SeekableByteChannel buildChannel(final String s) throws IOException {
             return FileChannel.open(Paths.get(this.mapNormalizedObjectNameToObjectName.get(s)), StandardOpenOption.READ);
         }
+
+        @Override
+        public Map<String, String> getMetadataValue(final String s) {
+            final Path path = Paths.get(this.mapNormalizedObjectNameToObjectName.get(s));
+            return getMetadataValues(path);
+        }
     }
 
+    private static Map<String, String> getMetadataValues(final Path path) {
+        try {
+            final FileTime lastModifiedTime = Files.getLastModifiedTime(path);
+            return ImmutableMap.of("ds3-last-modified", Long.toString(lastModifiedTime.toMillis()));
```

Comment:
you return an ImmutableMap, but don't indicate that in the function signature - why?